pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -828,6 +828,9 @@ cpdef disallow_ambiguous_unit(unit):
             "Units 'M', 'Y', and 'y' are no longer supported, as they do not "
             "represent unambiguous timedelta values durations."
         )
+    else:
+        unknown_kwargs = {"unit": unit}
+        raise ValueError(f"Unknown keyword arguments: {unknown_kwargs}")
```

Comment:
this will always raise.  the idea is to only raise for ambiguous units (Y, y, M), which this function already does